WordPress/src/main/java/org/wordpress/android/ui/mysite/MySiteStateProvider.kt

```diff
@@ -30,22 +31,26 @@ class MySiteStateProvider(
     }
 
     val state: LiveData<MySiteUiState> = selectedSiteRepository.siteSelected.switchMap { siteId ->
-        val result = MediatorLiveData<MySiteUiState>()
+        data class SiteIdToState(val siteId: Int?, val state: MySiteUiState = MySiteUiState()) {
+            fun update(partialState: PartialState): SiteIdToState {
+                return this.copy(state = state.update(partialState))
+            }
+        }
+        val result = MediatorLiveData<SiteIdToState>()
         val currentSources = if (siteId != null) {
             mySiteSources.map { source -> source.buildSource(siteId).distinctUntilChanged().asLiveData(bgDispatcher) }
         } else {
             mySiteSources.filterIsInstance(SiteIndependentSource::class.java)
                     .map { source -> source.buildSource().distinctUntilChanged().asLiveData(bgDispatcher) }
         }
-        result.value = MySiteUiState()
         for (newSource in currentSources) {
             result.addSource(newSource) { partialState ->
                 if (partialState != null) {
-                    result.value = (result.value ?: MySiteUiState()).update(partialState)
+                    result.value = (result.value ?: SiteIdToState(siteId)).update(partialState)
                 }
             }
         }
-        result
+        result.filter { it.siteId == null || it.state.site != null }.map { it.state }
```

Comment:
We might want to document this logic as it probably won't be obvious for everyone that reads it. What do you think?

Reply:
good point 👍